src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAnAotApp.cs

```diff
@@ -97,10 +97,51 @@ public void NativeAot_only_runs_when_switch_is_enabled(string targetFramework)
                 // The exe exist and should be native
                 File.Exists(publishedExe).Should().BeTrue();
                 IsNativeImage(publishedExe).Should().BeTrue();
+
+                var command = new RunExeCommand(Log, publishedExe)
+                    .Execute().Should().Pass()
+                    .And.HaveStdOutContaining("Hello world");                
```

Comment:
Not sure if here or in a different test, but it will be good to test that the warnings are generated even when the EnableAotAnalyzer is set to false and PublishAot is set to true.

Reply:
Created #25125 for this